### ISSUE TITLE
improve SMART parameters standardization

### ISSUE THREAD

**Issue by DominiqueMakowski** [easystats/effectsize#6]
imo this is a critical issue to be able to retrieve 'refit' standardized parameters with a posthoc method.

improvements are possible with especially for the case of interaction, but a more robust and systematic testing framework might be needed. also, a knowledge of model matrices when factors are involved appears as key.

**Comment by mattansb** [comment]
@DominiqueMakowski Can you explain what exactly the "smart" method is trying to do? It seems to break somewhat when there are formula-transformations (`log(y) ~ sqrt(x)`) and I'd like to try and fix that, but I don't know what I'm actually trying to achieve 😅 (Currently, methods "refit" and "basic" are the most stable, but that is only because I know what they're suppose to return...)

(I cases with transformations it will never be equal to method "refit" because the parameters themselves are estimated differently)

(As you mention above, this is also the issue with interactions - the centering changes the simple/conditional slope parameters, so it also cannot be the same as with "refit", no?)

So broadly asking, what would method "smart" do to this model:
```R
log(y) ~ sqrt(x) * some_factor
```

---

Also, what exactly is the conceptual difference between methods "smart" and "posthoc"?

**Comment by DominiqueMakowski** [comment]
TLDR; 

> So broadly asking, what would method "smart" do to this model:

No idea

Longer story:

so in simpler models basic and refit are equivalent. But for more complex models (especially with interactions, transformations etc) basic starts to depart from refit. IMO, refit gives the "gold standard" results, because they don't involve any posthoc transformation. However, the problem is that "refit" is computationally heavy (Bayes). So the goal of "smart" is to be a posthoc method (that does not refit the model from scratch but simply transforms the parameters with the info it has), but that gives the same results as "refit". 

So in simple cases, there should be no difference between the 3 methods basic, refit and smart. In more complex cases, smart aims at giving the same results as "refit". 

Basically "smart" is supposed to be "refit" minus the model refitting.  

So to get back to your initial question of what the result should be in that particular case, the expected result should be the same as the one given by "refit"...

---

About how it works, basically `standardize()` doesn't change factors, which are still booleans in the model matrix. So in order to adjust the parameters to mimic the "refit" method, for example sone must know if a given parameters refer to a continuous (in which case the parameter must be scaled by both the sd of the outcome and the predictor) or factor (only the outcome). I reckon there might be other cases where something particular should be made (reverse transformation when transformation are specified?)

**Comment by DominiqueMakowski** [comment]
but we can move slowly here, it doesn't need to be perfect from scratch, smart can always be equivalent to "basic" when we don't know how to retrieve the "refit"-like coefs. Basically it's like "basic+" method, i.e., it's fast, and will work in most cases, and in some cases it will need you the straightforward "basic" standardization
